irohad/network/peer_communication_stub.cpp

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright Soramitsu Co., Ltd. 2017 All Rights Reserved.
+ * http://soramitsu.co.jp
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <network/peer_communication_stub.hpp>
+
+namespace iroha {
+  namespace network {
+
+    using ametsuchi::Ametsuchi;
+    using validation::StatefulValidator;
+    using validation::ChainValidator;
+    using ordering::OrderingService;
+    using consensus::ConsensusService;
+    using dao::Block;
+
+    rxcpp::observable<rxcpp::observable<dao::Block>>
+    PeerCommunicationServiceStub::on_commit() {
```

Comment:
Yes, you are right. PeerCommunicationStub makes validation and write commit to Ametsuchi and is not depended on specific consensus.